src/dmd/lexer.d

```diff
@@ -2647,24 +2648,33 @@ private struct TimeStampInfo
 
     public static void initialize(const ref Loc loc) nothrow
     {
-        if (initdone)
+        // casting shared away because that will be needed if preview=restrictiveshared becomes the default
+        if (cast()initdone)
             return;
 
-        initdone = true;
-        time_t ct;
-        // https://issues.dlang.org/show_bug.cgi?id=20444
-        if (auto p = getenv("SOURCE_DATE_EPOCH"))
+        if (cas(&doing_init, false, true))
```

Comment:
```d
if (!cas(&doing_init, false, true))
{
    while(! atomicLoad(initdone)) {}
    return;
}
```

Reply:
yes good idea.
no need for the atomicLoad as only one thread can set the value.

Reply:
but better safe than sorry

Reply:
If you do omit the `atomicLoad` please put a comment explaining why the cast is fine. ditto for the start of the function